### PR TITLE
Custom Gentx

### PR DESCRIPTION
**Pull request by jackzampolin**
This PR changes the gentx functionality to require that validators have their ETH key at network startup.

### CODE REVIEW

File: module/cmd/peggy/cmd/gentx.go [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+		Short: "Generate a genesis tx carrying a self delegation, oracle key delegation and orchestrator key delegation",
+		Args:  cobra.ExactArgs(4),
+		Long: fmt.Sprintf(`Generate a genesis transaction that creates a validator with a self-delegation, oracle key 
+delegation and orchestrator key delegation that is signed by the key in the Keyring referenced by a given name. A node 
+ID and Bech32 consensus pubkey may optionally be provided. If they are omitted, they will be retrieved from the 
+priv_validator.json file. The following default parameters are included:
+    %s
+
+Example:
+$ %s gentx my-key-name 1000000stake --home=/path/to/home/dir --keyring-backend=os --chain-id=test-chain-1 \
```
Comment (fedekunze) [inline module/cmd/peggy/cmd/gentx.go:56]: update example

File: module/cmd/peggy/cmd/gentx.go [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+
+			name := args[0]
+			key, err := clientCtx.Keyring.Key(name)
+			if err != nil {
+				return errors.Wrapf(err, "failed to fetch '%s' from the keyring", name)
+			}
+
+			ethAddress := args[2]
+
+			// TODO: eth address validation
```
Comment (fedekunze) [inline module/cmd/peggy/cmd/gentx.go:119]: there's a util function on the peggy module

File: module/cmd/peggy/cmd/gentx.go [hunk 3]
```diff
@@ -281,246 +277,3 @@ func writeSignedGenTx(clientCtx client.Context, outputDocument string, tx sdk.Tx
...
-			return appGenTxs, persistentPeers, fmt.Errorf("expected TxWithMemo, got %T", genTx)
-		}
-		nodeAddrIP := memoTx.GetMemo()
-		if len(nodeAddrIP) == 0 {
-			return appGenTxs, persistentPeers, fmt.Errorf("failed to find node's address and IP in %s", fo.Name())
-		}
-
-		// genesis transactions must be single-message
-		msgs := genTx.GetMsgs()
-		// if len(msgs) != 1 {
```
Comment (jackzampolin) [inline module/cmd/peggy/cmd/gentx.go]: So the one issue here is that collect gentxs currently errors when `len(msgs) != 1`. Also I think we may want to add some other custom validation for gentxs that will help us start chains so I think we should leave this.

Comment (jackzampolin) [reply module/cmd/peggy/cmd/gentx.go]: cc @fedekunze

**Comment by jkilpatr** [comment]
working on integrating this into the tests now 👍🏻

**Comment by jackzampolin** [comment]
Awesome @jkilpatr

**Comment by jkilpatr** [comment]
rebased and added my own patch in #308

**Pull request closed without merging**